### PR TITLE
Generate a session id for new sessions with data

### PR DESCRIPTION
**Pull request by geerteltink**
If a new session was generated, the session id isn't stored. On persistence it doesn't (re)generate an id which is expected. You only need a session if there is data to store. However there is never a check performed for changed data session.

This PR tries to fix that and checks if session data is changed and generates a session id if there wasn't one. This makes sure that data is stored if there wasn't a session yet.

### CODE REVIEW

File: src/PhpSessionPersistence.php [hunk 1]
```diff
@@ -100,7 +99,10 @@ public function initializeSessionFromRequest(ServerRequestInterface $request) :
 
     public function persistSession(SessionInterface $session, ResponseInterface $response) : ResponseInterface
     {
-        if ($session->isRegenerated()) {
+        // Regenerate if the session is marked as regenerated
+        // Regenerate if there is no cookie id set but the session has changed (new session with data)
+        if ($session->isRegenerated()
+            || ($session->hasChanged() && ! $this->cookie)) {
```
Comment (michalbundyra) [inline src/PhpSessionPersistence.php]: `) {` from `if` statement should be in new line

File: test/PhpSessionPersistenceTest.php [hunk 2]
```diff
@@ -422,4 +422,30 @@ public function testPersistSessionReturnsExpectedResponseWithoutAddedCacheHeader
 
         $this->restoreOriginalSessionIniSettings($ini);
     }
+
+    public function testCookiesNotSetWithoutRegenerate(): void
```
Comment (michalbundyra) [inline test/PhpSessionPersistenceTest.php]: Usually we are not adding `: void` in tests, but if we keep it here, there should be one space before `:` (consistency).

File: test/PhpSessionPersistenceTest.php [hunk 3]
```diff
@@ -422,4 +422,30 @@ public function testPersistSessionReturnsExpectedResponseWithoutAddedCacheHeader
...
+    {
+        $persistence = new PhpSessionPersistence();
+        $request = new ServerRequest();
+        $session = $persistence->initializeSessionFromRequest($request);
+
+        $response = new Response();
+        $response = $persistence->persistSession($session, $response);
+
+        $this->assertEmpty($response->getHeaderLine('Set-Cookie'));
```
Comment (michalbundyra) [inline test/PhpSessionPersistenceTest.php]: Maybe we can just check `$this->assertFalse($response->hasHeader())` ?

**Comment by pine3ree** [comment]
Hello @xtreamwayz 
I would just put `! $this->cookie` before the `hasChanged` test to avoid function call (I know, I know micro-opt... :-) ) . kind regards.

**Comment by geerteltink** [comment]
Incorporated changes from @webimpress and @pine3ree.

File: src/PhpSessionPersistence.php [hunk 4]
```diff
@@ -16,7 +15,6 @@
 use Zend\Expressive\Session\Session;
 use Zend\Expressive\Session\SessionInterface;
 use Zend\Expressive\Session\SessionPersistenceInterface;
-
```
Comment (michalbundyra) [inline src/PhpSessionPersistence.php]: Revert this change, we should have an empty line between different type of imports.

**Comment by weierophinney** [comment]
Thanks, @xtreamwayz!
